pkg/cmd/repo/create/create.go

```diff
@@ -52,66 +63,70 @@ func NewCmdCreate(f *cmdutil.Factory, runF func(*CreateOptions) error) *cobra.Co
 
-			To create a repository non-interactively, supply the following:
-			- the name argument;
-			- the %[1]s--confirm%[1]s flag;
-			- one of %[1]s--public%[1]s, %[1]s--private%[1]s, or %[1]s--internal%[1]s.
+			To create a new remote repository, supply the repository name 
+			and one of %[1]s--public%[1]s, %[1]s--private%[1]s, or %[1]s--internal%[1]s.
+			Toggle %[1]s--clone%[1]s to clone the new repository locally.
 
-			To toggle off %[1]s--enable-issues%[1]s or %[1]s--enable-wiki%[1]s, which are enabled
-			by default, use the %[1]s--enable-issues=false%[1]s syntax.
+			To create a remote repository from an existing local repository, 
+			specify the source directory with %[1]s--source%[1]s.
 		`, "`"),
 		Args: cobra.MaximumNArgs(1),
-		Example: heredoc.Doc(`
-			# create a repository under your account using the current directory name
-			$ git init my-project
-			$ cd my-project
-			$ gh repo create
-
-			# create a repository with a specific name
-			$ gh repo create my-project
-
-			# create a repository in an organization
-			$ gh repo create cli/my-project
-
-			# disable issues and wiki
-			$ gh repo create --enable-issues=false --enable-wiki=false
-	  `),
-		Annotations: map[string]string{
-			"help:arguments": heredoc.Doc(`
-				A repository can be supplied as an argument in any of the following formats:
-				- "OWNER/REPO"
-				- by URL, e.g. "https://github.com/OWNER/REPO"
-			`),
-		},
 		RunE: func(cmd *cobra.Command, args []string) error {
 			if len(args) > 0 {
 				opts.Name = args[0]
 			}
 
-			if len(args) == 0 && (opts.GitIgnoreTemplate != "" || opts.LicenseTemplate != "") {
-				return cmdutil.FlagErrorf(".gitignore and license templates are added only when a specific repository name is passed")
+			if len(args) == 0 && cmd.Flags().NFlag() == 0 {
+				if !opts.IO.CanPrompt() {
+					return cmdutil.FlagErrorf("at least one argument required in non-interactive mode")
+				}
+				opts.Interactive = true
+			} else {
+				// exactly one visibility flag required
+				if !opts.Public && !opts.Private && !opts.Internal {
+					return cmdutil.FlagErrorf("`--public`, `--private`, or `--internal` required when not running interactively")
+				}
+				err := cmdutil.MutuallyExclusive(
+					"expected exactly one of `--public`, `--private`, or `--internal`",
+					opts.Public, opts.Private, opts.Internal)
+				if err != nil {
+					return err
+				}
+
+				if opts.Public {
+					opts.Visibility = "PUBLIC"
+				} else if opts.Private {
+					opts.Visibility = "PRIVATE"
+				} else {
+					opts.Visibility = "INTERNAL"
+				}
 			}
 
-			if opts.Template != "" && (opts.GitIgnoreTemplate != "" || opts.LicenseTemplate != "") {
-				return cmdutil.FlagErrorf(".gitignore and license templates are not added when template is provided")
+			if opts.Source != "" && (opts.Clone || opts.GitIgnoreTemplate != "" || opts.LicenseTemplate != "") {
+				// which options allowed with --source?
```

Comment:
`--remote` and `--push` work if and only if `--source` is used

the ones you have listed here don't work with `--source` (as you are enforcing)

these are allowed with or without `--source`:

* `--homepage`
* `--team`
* `--description`
* `--private`
* `--public`
* `--internal`
* `--homepage`
* `--disable-issues`
* `--disable-wiki`